kubernetes-go-guestbook/simple/main.go

```diff
@@ -0,0 +1,264 @@
+					Spec: &corev1.PodSpecArgs{
+						Containers: corev1.ContainerArray{redisFollowerContainer},
+					},
+				},
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Redis follower Service
+		_, err = corev1.NewService(ctx, "redis-follower", &corev1.ServiceArgs{
+			Metadata: &metav1.ObjectMetaArgs{
+				Annotations: pulumi.StringMap{"foo": pulumi.String("baa")},
+			},
+			Spec: &corev1.ServiceSpecArgs{
+				Ports: corev1.ServicePortArray{
+					corev1.ServicePortArgs{
+						Port:       pulumi.IntPtr(6379),
+						TargetPort: pulumi.Any(6379),
+					},
+				},
+				Selector: redisFollowerLabels,
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		frontendContainer := corev1.ContainerArgs{
+			Name:  pulumi.String("php-redis"),
+			Image: pulumi.String("gcr.io/google-samples/gb-frontend:v4"),
+			Resources: &corev1.ResourceRequirementsArgs{
+				Requests: pulumi.StringMap{
+					"cpu":    pulumi.String("100m"),
+					"memory": pulumi.String("100Mi"),
+				},
+			},
+			Env: corev1.EnvVarArray{
+				corev1.EnvVarArgs{
+					Name:  pulumi.StringPtr("GET_HOSTS_FROM"),
+					Value: pulumi.StringPtr("dns"),
+				},
+			},
+			Ports: corev1.ContainerPortArray{
+				&corev1.ContainerPortArgs{
+					ContainerPort: pulumi.IntPtr(80),
+				},
+			},
+		}
+
+		// Frontend Deployment
+		_, err = appsv1.NewDeployment(ctx, "frontend", &appsv1.DeploymentArgs{
+			Spec: appsv1.DeploymentSpecArgs{
+				Selector: &metav1.LabelSelectorArgs{
+					MatchLabels: frontendLabels,
+				},
+				Replicas: pulumi.IntPtr(1),
+				Template: &corev1.PodTemplateSpecArgs{
+					Metadata: &metav1.ObjectMetaArgs{
+						Labels: frontendLabels,
+					},
+					Spec: &corev1.PodSpecArgs{
+						Containers: corev1.ContainerArray{frontendContainer},
+					},
+				},
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Frontend Service
+		var frontendServiceType string
+		if isMinikube {
+			frontendServiceType = "ClusterIP"
+		} else {
+			frontendServiceType = "LoadBalancer"
+		}
+		frontendService, err := corev1.NewService(ctx, "frontend", &corev1.ServiceArgs{
+			Spec: &corev1.ServiceSpecArgs{
+				Type: pulumi.StringPtr(frontendServiceType),
+				Ports: corev1.ServicePortArray{
+					corev1.ServicePortArgs{
+						Port: pulumi.IntPtr(80),
+					},
+				},
+				Selector: frontendLabels,
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Export frontend Service IP
+		if isMinikube {
+			ctx.Export("frontendIP", frontendService.Spec.ApplyT(
+				func(spec *corev1.ServiceSpec) string {
+					if spec.ClusterIP == nil {
+						return ""
+					}
+					return *spec.ClusterIP
+				}))
+		} else {
+			ctx.Export("frontendIP", frontendService.Status.ApplyT(
+				func(status *corev1.ServiceStatus) string {
+					ingress := status.LoadBalancer.Ingress[0]
+					if ingress.Hostname != nil {
+						ip := *ingress.Hostname
+						if len(ip) > 0 {
+							return ip
+						}
+					}
+
+					if ingress.Ip != nil {
+						return *ingress.Ip
+					}
+
+					return ""
```

Comment:
Total nit - but perhaps make the polarity of this match the other branch above?

Reply:
Will try to simplify this